src/components/overlay/index.js

```diff
@@ -0,0 +1,111 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet, Image} from 'react-native';
+import {Colors} from '../../style';
+import {PureBaseComponent} from '../../commons';
+import View from '../view';
+
+
+const OVERLY_TYPES = {
+  VERTICAL: 'vertical',
+  TOP: 'top',
+  BOTTOM: 'bottom',
+  SOLID: 'solid'
+}
+const gradientImage = require('./assets/GradientOverlay.png');
+
+/**
+ * @description: Overlay view with types (default, top, bottom, solid)
+ * @extends: Image
+ * @extendsLink: https://facebook.github.io/react-native/docs/image
+ */
+export default class Overlay extends PureBaseComponent {
+  static displayName = 'Overlay';
+
+  static propTypes = {
+    /** 
+    * The type of overlay to set on top of the image 
+    */
+    type: PropTypes.oneOf(_.values(OVERLY_TYPES))
+  };
+
+  static overlayTypes = OVERLY_TYPES;
+
+  getStyleByType() {
+    switch (this.props.type) {
+      case OVERLY_TYPES.TOP: 
+        return styles.top;
+      case OVERLY_TYPES.BOTTOM: 
+        return styles.bottom;
+      case OVERLY_TYPES.SOLID: 
+        return styles.solid
+      default:
+        break;
+    }
+  }
+
+  getAttributeStyles(style, attName) {
+    const flatten = StyleSheet.flatten(style);
+    let attributes;
+
+    if (flatten) {
+      attributes = _.chain(flatten)
+      .pickBy((value, key) => _.includes(key, attName))
+      .value();
+    }
+    return attributes;
+  }
+
+  renderImage(typeStyle) {
+    const {type, style} = this.props;
+    const image = type !== OVERLY_TYPES.SOLID ? gradientImage : undefined;
+    const marginStyle = type !== OVERLY_TYPES.VERTICAL ? this.getAttributeStyles(style, 'margin') : undefined;
```

Comment:
Just so I understand, for what use case is it?

Reply:
Since the overlay image is a sibling to the image it should get the margin style sent to the image in over to layout correctly. In case of the VERTICAL type the margin should go to the container of the two overlay images and not to the overlay image itself.